bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/mascot/Mascot.scala

```diff
@@ -77,23 +81,29 @@ object Mascot {
    * Finds the best asterisms for the given list of stars.
    * @param starList unfiltered list of stars from a catalog query
    * @param factor multiply strehl min, max and average by this value (depends on instrument filter: See REL-426)
-   * @param filter a filter function that returns false if the Star should be excluded
    * @param progress a function(strehl, count, total) called for each asterism as it is calculated
+   * @param filter a filter function that returns false if the Star should be excluded
+   * @param asterismPreFilter a filter function that returns false if a candidate asterism should be excluded
    * @return a tuple: (list of stars actually used, list of asterisms found)
    */
   def findBestAsterism(starList: List[Star],
                        factor: Double = defaultFactor,
                        progress: ProgressFunction = defaultProgress,
-                       filter: Star => Boolean = defaultFilter)
+                       filter: Star => Boolean = defaultFilter,
+                       asterismPreFilter: List[SiderealTarget] => Boolean = defaultAsterismPreFilter)
```

Comment:
Should the argument of this filter be `Asterism` (i don't know if that exists)?

Reply:
Looking at the `Asterism` class, I'm not sure we have a full implementation yet that allows construction of asterisms with more than a single `SPTarget`, unless I'm misunderstanding... but yes, that might be a better way to go in the long run through all of MASCOT if we can do it.